tools/myst/host/syscall.c

```diff
@@ -422,3 +423,29 @@ long myst_utimensat_ocall(
     SAVE_CALL_RESTORE_IDENTITY_RETURN(
         uid, gid, syscall(SYS_utimensat, dirfd, pathname, times, flags));
 }
+
+long myst_sched_setaffinity_ocall(
+    pid_t pid,
+    size_t cpusetsize,
+    const struct myst_cpu_set* mask)
+{
+    RETURN(syscall(SYS_sched_setaffinity, pid, cpusetsize, mask));
+}
+
+MYST_STATIC_ASSERT(sizeof(struct myst_cpu_set) == sizeof(cpu_set_t));
+
+long myst_sched_getaffinity_ocall(
+    pid_t pid,
+    size_t cpusetsize,
+    struct myst_cpu_set* mask)
+{
+    RETURN(syscall(SYS_sched_getaffinity, pid, cpusetsize, mask));
```

Comment:
The implementation returns non-zero value on successes. On a 4 core system, I saw return value of 8, with the mask showing 4 cores. Changing the implementation to call libc sched_getaffinity() instead, the return value is 0.

Reply:
I see it too. Still trying to fix.

Reply:
Okay, the man page says that ``sched_getaffinity`` returns the number of bytes written to the mask, which is different than the number of CPUs.

```
On success, the raw sched_getaffinity() system call returns the
       number of bytes placed copied into the mask buffer; this will be
       the minimum of cpusetsize and the size (in bytes) of the
       cpumask_t data type that is used internally by the kernel to
       represent the CPU set bit mask.
```

Reply:
The ``sched_getaffinity`` ocall needed to be fixed to set ``size=cpusetsize``. I added an overwrite test (that failed initially) and then fixed it.